src/main/java/graphql/execution/instrumentation/ChainedInstrumentation.java

```diff
@@ -175,6 +179,18 @@ public ExecutionStrategyInstrumentationContext beginExecutionStrategy(Instrument
         return new ChainedExecutionStrategyInstrumentationContext(mapAndDropNulls(instrumentations, mapper));
     }
 
+    @ExperimentalApi
+    @Override
+    public InstrumentationContext<ExecutionResult> beginDeferredField(InstrumentationDeferredFieldParameters parameters, InstrumentationState instrumentationState) {
+        return new ChainedDeferredExecutionStrategyInstrumentationContext(instrumentations.stream()
+                .map(instrumentation -> {
+                    InstrumentationState specificState = getSpecificState(instrumentation, instrumentationState);
+                    return instrumentation.beginDeferredField(parameters, specificState);
+                })
+                .collect(Collectors.toList()));
+
+    }
```

Comment:
I have another PR that gets rid of returning `ExecutionResult` from fields - turned out we never used the ER per se but just its data value.  So what can we do here to avoid re-introducing it. 

I am noting this now as a source of friction with other work.

Reply:
https://github.com/graphql-java/graphql-java/pull/3330

I am likely to merge soon so this will conflict in design

Reply:
eg

```
    @Deprecated(since="2023-09-11" )
    @Nullable
    default InstrumentationContext<ExecutionResult> beginField(InstrumentationFieldParameters parameters, InstrumentationState state) {
        return beginField(parameters.withNewState(state));
    }
```

Reply:
Thanks for the heads up ;-)
I'll wait for your PR to be merged and resolve conflicts - no biggie